src/actions/tools/RectangleDrawTool.js

```diff
@@ -0,0 +1,82 @@
+import Tool from "./Tool";
+import { createSvgFromPoints } from "../../utils";
+
+class RectangleDrawTool extends Tool {
+  constructor(model) {
+    super(model);
+    this.toolName = 'RECTANGLE';
+  }
+
+  reset() {
+    super.reset();
+    this.model.annotations.pop();
+  }
+
+  onPress( action ) {
+    if (!this.model.controlsactive) {
+      return;
+    }
+    if (this.model.activityInProgress) {
+      return;
+    }
+
```

Comment:
above two checks are different from OnPress of the other two tools. Please try to follow similar code patterns

Reply:
Removed the early returns